buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantExtension.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.vagrant;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.util.Map;
+
+public class VagrantExtension {
+
+    private final Property<String> box;
+    private final MapProperty<String, Object> hostEnv;
+    private final MapProperty<String, Object> vmEnv;
+    private final RegularFileProperty vagrantfile;
+    private boolean isWindowsVM;
+
+    public VagrantExtension(Project project) {
+        this.box = getObjectFactory().property(String.class);
+        this.hostEnv = getObjectFactory().mapProperty(String.class, Object.class);
+        this.vmEnv = getObjectFactory().mapProperty(String.class, Object.class);
+        this.vagrantfile = getObjectFactory().fileProperty();
+        this.vagrantfile.convention(project.getRootProject().getLayout().getProjectDirectory().file("Vagrantfile"));
+        this.isWindowsVM = false;
+    }
+
+    @Inject
+    protected ObjectFactory getObjectFactory() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Input
```

Comment:
Annotations will have no affect on extensions

Reply:
Ah yes, I need to cleanup the inputs for all the changes here in general. I think the idea here was to make the VagrantCommandTask expose this extension as a NestedInput, in which case I believe inputs do matter here.

Reply:
They are if they are a task property annotated with `@Nested` although it is not in this case.